pkg/gofr/context_test.go

```diff
@@ -36,3 +38,55 @@ func Test_newContextSuccess(t *testing.T) {
 	assert.Equal(t, map[string]string{"key": "value"}, body, "TEST Failed \n unable to read body")
 	require.NoError(t, err, "TEST Failed \n unable to read body")
 }
+
+func TestContext_AddTrace(t *testing.T) {
+	ctx := Context{
+		Context: context.Background(),
+	}
+
+	span := ctx.Trace("Some Work")
+
+	defer span.End()
+
+	require.NotEqual(t, ctx.Context, context.Background())
```

Comment:
```suggestion
	ctxBase := context.Background()
	ctx := Context{
		Context: ctxBase,
	}

	span := ctx.Trace("Some Work")

	defer span.End()

	assert.NotEqual(t, ctxBase, ctx.Context)
```

- used a baseCtx because context.Background creates a new context.
- used assert because, there is no need to use require
-fixed inverted actual expected in NotEqual